### DISCUSSION TITLE
v1.0.0-rc4

### DISCUSSION THREAD

**Discussion by Mrs4s** [Mrs4s/go-cqhttp#1825] [General]
## Changelog
新年快乐🎉
此版本修复了很多问题.
如果此次更新后存在无法登录的情况, 请删除 `device.json` 重试.
关于不同网络环境下登录的滑条验证问题, 在弄了! 下个版本应该会有解决方案. 现在请尽量在同一网络环境登录

### 新增
c84d583 数据库新增了了 `sqlite3` 的可选支持.  @fumiama 
ec4b3cc 在 `Windows` 下的提示弹窗支持了高分辨率. @Ink-33 
bae00b5 `send_forward_msg` API 会返回 `forward id` 了. @Mrs4s 
933bdee 对群消息发送失败的返回值进行了优化, 现在API返回会提示具体错误了. @Mrs4s 
93074ef 支持手动输入滑条验证码的 `ticket` 了. @shigma 
36bf579 新增删除群公告`_del_group_notice` API. @Akegarasu 
7707e78 `get_stranger_info` API 新增 `vip_level` 字段. @wdvxdr1123 
b981b45 新增对输入环境的tty检查. @shigma 
b05a64f 邀请入群新增 `invitor_id` 字段. @xiangxiangxiong9 

### 逻辑修复
a0fba6a 修复 `send_private_forward_msg` 发送的消息无法在苹果客户端查看的问题. @Akegarasu 
7e75035 修复 `send_private_forward_msg` 发送缓存后的图片无法再 Windows 客户端上显示的问题. @Akegarasu 
f772996 修复 `send_private_forward_msg` 不支持直接引用他人合并转发的问题. @Mrs4s 
0248c86 修复特定消息可能会导致反向WS发送空数据包导致接收端断开连接的问题. @Mrs4s 
93fa360 修复一处 `daemon` 判断错误导致无法正确启动的问题. @sgpublic 
069a764 修复部分情况下私聊回复的 `reply_id` 不存在的问题. @Akegarasu 
4ff6121 修复 `-w` 与 `-d` 一起使用时会输出错误的 `pid` 的问题. @sgpublic 
140192c 修复当账号只有一个群时发送合并转发会出错的问题 @Mrs4s 

### 协议修复
修复语音消息无法接收的问题. @super1207
修复部分频道消息会解析失败的问题. @Mrs4s
修复读取私聊合并转发内容会出错的问题. @Azure1160
修复群文件操作失败会报错的问题. @MingxuanGame

### 杂项更新
为账号被风控无法发送群消息加入了单独的提示. @Mrs4s
为 `iPad` `MacOS` `AndroidPhone` 协议更新了版本.
支持了腾讯新更新的 `aPad` 协议. 
移除了登录时滑条验证码的自动选择.
将 `get_version_info` 的 `protocol` 字段替换为 `protocol_name` 字段
移除了 `btree` 的使用, 以解决奇怪的错误

<hr /><em>This discussion was created from the release <a href='https://github.com/Mrs4s/go-cqhttp/releases/tag/v1.0.0-rc4'>v1.0.0-rc4</a>.</em>

**Comment by ttxs163** [comment]
bug反馈：群内某人被移出群测试 如在极短时间内连续移出多个人时，无法监测到全部被移出者的QQ
